lib/hls/hls_parser.js

```diff
@@ -4179,15 +4214,50 @@ shaka.hls.HlsParser = class {
       return null;
     }
 
+    const keyUris = shaka.hls.Utils.constructSegmentUris(
+        getUris(), drmTag.getRequiredAttrValue('URI'), variables);
+
+    const requestType = shaka.net.NetworkingEngine.RequestType.KEY;
+    const request = shaka.net.NetworkingEngine.makeRequest(
+        keyUris, this.config_.retryParameters);
+    const keyResponse =
+        await this.makeNetworkRequest_(request, requestType);
```

Comment:
If I'm understanding correctly, since we call `parseDrmInfo_` inside `updateStream_`, this means that in a livestream with identity format we will re-request the key each time we update the manifest.

We determine if we need to replace the drm infos in the stream when updating by comparing the keys, so we probably shouldn't just cache the results or something in case they actually did change.

Not a huge deal or anything, just an observation.

Reply:
> If I'm understanding correctly, since we call parseDrmInfo_ inside updateStream_, this means that in a livestream with identity format we will re-request the key each time we update the manifest.

This seems bad.  I'd rather not have a big performance problem on live if we can fix it now.

> We determine if we need to replace the drm infos in the stream when updating by comparing the keys, so we probably shouldn't just cache the results or something in case they actually did change.

They can't change.  That doesn't make sense.  A cache seems like a good idea.

Reply:
Cache added!